x-pack/plugins/observability_solution/slo/public/embeddable/slo/error_budget/types.ts

```diff
@@ -4,13 +4,30 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import { EmbeddableInput } from '@kbn/embeddable-plugin/public';
+import { SerializedTitles } from '@kbn/presentation-publishing';
+import { DefaultEmbeddableApi } from '@kbn/embeddable-plugin/public';
 import { Subject } from 'rxjs';
+import {
+  type CoreStart,
+  IUiSettingsClient,
+  ApplicationStart,
+  NotificationsStart,
+} from '@kbn/core/public';
 
 export interface EmbeddableSloProps {
   sloId: string | undefined;
   sloInstanceId: string | undefined;
   reloadSubject?: Subject<boolean>;
   onRenderComplete?: () => void;
 }
-export type SloErrorBudgetEmbeddableInput = EmbeddableInput & EmbeddableSloProps;
+
+export type SloErrorBudgetEmbeddableState = SerializedTitles & EmbeddableSloProps;
```

Comment:
nit: I wouldn't use the whole `EmbeddableSloProps` type here for your serialized state. This should _only_ be what you intend to **save** into the Dashboard. Likely just the `sloId` and the `sloInstanceId`.

Reply:
This looks much better, thank you!